awx/ui/client/src/notifications/notificationTemplates.form.js

```diff
@@ -333,6 +333,50 @@ export default ['i18n', function(i18n) {
                 subForm: 'typeSubForm',
                 ngDisabled: '!(notification_template.summary_fields.user_capabilities.edit || canAdd)'
             },
+            mattermost_url: {
+                label: i18n._('Target URL'),
+                type: 'text',
+                awRequiredWhen: {
+                    reqExpression: "mattermost_required",
+                    init: "false"
+                },
+                ngShow: "notification_type.value == 'mattermost' ",
+                subForm: 'typeSubForm',
+                ngDisabled: '!(notification_template.summary_fields.user_capabilities.edit || canAdd)'
+            },
+            mattermost_username: {
+                label: i18n._('Username'),
+                type: 'text',
+                awRequiredWhen: {
```

Comment:
If username is not required then you can remove this `awRequiredWhen` block.  The form generator will generate a field that's not required by default.  The same is true for channel and icon url.